src/providers/wms/qgswmssourceselect.cpp

```diff
@@ -1020,6 +1023,39 @@ void QgsWMSSourceSelect::collectSelectedLayers( QStringList &layers, QStringList
   }
 }
 
+void QgsWMSSourceSelect::collectDimensions( QStringList &layers, QgsDataSourceUri &uri )
+{
+  for ( const QgsWmsLayerProperty layerProperty : mLayerProperties )
+  {
+    if ( layerProperty.name == layers.join( QStringLiteral( "," ) ) )
+    {
+      // Check for layer dimensions
+      for ( const QgsWmsDimensionProperty &dimension : qgis::as_const( layerProperty.dimensions ) )
+      {
+        // add temporal dimensions only
+        if ( dimension.name == QLatin1String( "time" ) ||
+             dimension.name == QLatin1String( "reference_time" ) )
+        {
+          QString name = dimension.name == QLatin1String( "time" ) ?
+                         QString( "timeDimensionExtent" ) : QString( "referenceTimeDimensionExtent" );
```

Comment:
```suggestion
                         QStringLiteral( "timeDimensionExtent" ) : QStringLiteral( "referenceTimeDimensionExtent" );
```

(apparently qt 6 is going to fix this mess and avoid the confusing need for QStringLiteral and QLatin1String. Fingers crossed!!)

Reply:
That will help for sure!